### PR TITLE
feat: announce: reddit

### LINKED ISSUE #2424: add Reddit as a new GoReleaser Announce
**Is your feature request related to a problem? Please describe.**
It is not related to a problem, it is a new feature.

**Describe the solution you'd like**
We (w/@Dentrax) would like to add another integration (Reddit in this case) to the [GoReleaser Announce](https://goreleaser.com/customization/announce/) feature.

Reddit also has API docs which you can see from [here](https://www.reddit.com/dev/api). So, we can easily add this support to the GoReleaser Announce feature.

### PR DESCRIPTION
**Pull request by developer-guy**
Signed-off-by: Batuhan Apaydın <batuhan.apaydin@trendyol.com>

add Reddit support to the Announces, and fix #2424 

to add Reddit support

https://www.reddit.com/r/test/comments/pbtik1/goreleaser_v1123_was_just_released_get_it_while/
https://github.com/developer-guy/goreleaser/runs/3429508179?check_suite_focus=true

### CODE REVIEW

**Comment by caarlos0** [comment]
> https://www.reddit.com/r/test/comments/pbtik1/goreleaser_v1123_was_just_released_get_it_while/

seems like the link was added to the title? 🤔

**Comment by developer-guy** [comment]
> > https://www.reddit.com/r/test/comments/pbtik1/goreleaser_v1123_was_just_released_get_it_while/
> 
> 
> 
> seems like the link was added to the title? 🤔

This is from our first test before your reviews; we have just wanted to see the working result 👌 @caarlos0

File: .goreleaser.yml [hunk 1]
```diff
@@ -202,3 +202,8 @@ announce:
   twitter:
     enabled: true
     message_template: 'GoReleaser {{ .Tag }} was just released! Get it while its fresh: https://github.com/goreleaser/goreleaser/releases/tag/{{ .Tag }}'
+  reddit:
+    enabled: true
+    title_template: 'GoReleaser {{ .Tag }} was just released!'
+    url_template: 'https://github.com/goreleaser/goreleaser/releases/tag/{{ .Tag }}'
+    sub: test
```
[the remark this answers is not in the document]
Comment (caarlos0) [reply .goreleaser.yml:209]:
```suggestion
```

File: internal/pipe/announce/announce.go [hunk 2]
```diff
@@ -4,6 +4,8 @@ package announce
 import (
 	"fmt"
 
+	"github.com/goreleaser/goreleaser/internal/pipe/reddit"
+
```
Comment (caarlos0) [inline internal/pipe/announce/announce.go:8]:
```suggestion
```

File: pkg/defaults/defaults.go [hunk 3]
```diff
@@ -5,6 +5,8 @@ package defaults
 import (
 	"fmt"
 
+	"github.com/goreleaser/goreleaser/internal/pipe/reddit"
+
```
Comment (caarlos0) [inline pkg/defaults/defaults.go:9]:
```suggestion
```

**Comment by caarlos0** [comment]
Thanks!

**Comment by developer-guy** [comment]
Thanks for merging @caarlos0 ❤️

**Pull request merged**